src/parser/types.rs

```diff
@@ -195,12 +200,16 @@ pub enum TyKind {
     // U16,
     // U32,
     // U64,
+
+    /// A generic type for a numeric variable.
+    Generic(String),
```

Comment:
`TyKind::BigInt` doesn't contain the actual value, so wondering why `Generic` does here

Reply:
It is to represent a generic variable with its name, which can be checked against the generic definition of a function signature.

Reply:
Right, this is for the case `fn create_arr(nn: NN)`. The TyKind::Generic is to represent the `NN` in this case.

We are going to change it to `fn create_arr(const NN)`. `is_generic_parameter` can be used to determine if the `NN` is a generic parameter.

So `TyKind::Generic` will be removed.